### PR TITLE
feat(kc): add configurable auth path for keycloak requests

### PR DESCRIPTION
**Pull request by Phil91**
## Description

Make the auth path of keycloak configureable

## Why

With the keycloak upgrade the /auth path in the urls of keycloak was removed, to be able to still use older versions its now configurable

## Issue

N/A - Jira Issue: CPLP-3214

## Depending CD PR
[#95](https://github.com/eclipse-tractusx/portal-cd/pull/95)

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/keycloak/Keycloak.Factory/KeycloakSettingData.cs [hunk 1]
```diff
@@ -27,19 +27,20 @@ namespace Org.Eclipse.TractusX.Portal.Backend.Keycloak.Factory;
...
     public string? User { get; set; }
+
     public string? Password { get; set; }
+
     public string? ClientId { get; set; }
+
     public string? ClientSecret { get; set; }
+
     public string? AuthRealm { get; set; }
+    public bool UseAuthTrail { get; set; }
```
Comment (evegufy) [inline src/keycloak/Keycloak.Factory/KeycloakSettingData.cs:41]: Looks good! could you please add the setting in every relevant appsettings.json file with default value 'false'.

**Review by evegufy — CHANGES_REQUESTED** [review]

File: src/keycloak/Keycloak.Factory/KeycloakSettingData.cs (hunk 1 above)
Comment (Phil91) [reply src/keycloak/Keycloak.Factory/KeycloakSettingData.cs:41]: sure, done.

Comment (evegufy) [reply src/keycloak/Keycloak.Factory/KeycloakSettingData.cs:41]: Thanks! I re-thought our conversation from earlier: if you create a PR for the helm chart adding this setting with value 'true', it shouldn't be breaking change for dev, could you please also create that PR?
Also there's a code smell, could you handle that?

Comment (Phil91) [reply src/keycloak/Keycloak.Factory/KeycloakSettingData.cs:41]: will take care about the pr.
regarding the code smell nothing really we could do against it, we need the parameters in the method. but i'll add a comment in sonar

Comment (Phil91) [reply src/keycloak/Keycloak.Factory/KeycloakSettingData.cs:41]: CD [PR](https://github.com/eclipse-tractusx/portal-cd/pull/95) has been created. I've added a comment for the code smell in sonar

**Review by evegufy — APPROVED** [review]

**Pull request merged**